src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +101,196 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                    amount: {
+                        value: parseFloat(String(option.cost)).toFixed(2),
+                        currency_code: 'USD',
+                    },
+                };
+            });
+
+        shippingOptions?.sort( (a, b) => {
+            return (a.selected === b.selected) ? 0 : a ? -1 : 1;
+        });
+        if (shippingOptions && shippingOptions[0].id) {
+            this._shippingOptionId = shippingOptions[0].id;
+        }
+        this._submittedShippingAddress = this._currentShippingAddress;
+
+        if (shippingOptions) {
+            actions.order.patch([
+                {
+                    op: 'replace',
+                    path: '/purchase_units/@reference_id==\'default\'/amount',
+                    value: {
+                        currency_code: 'USD',
+                        value: (parseFloat(String(baseOrderAmount)) + parseFloat(shippingAmount)).toFixed(2),
+                        breakdown: {
+                            item_total: {
+                                currency_code: 'USD',
+                                value: baseOrderAmount,
+                            },
+                            shipping: {
+                                currency_code: 'USD',
+                                value: shippingAmount,
+                            },
+                        },
+                    },
+                },
+                {
+                    op: 'add',
+                    path: '/purchase_units/@reference_id==\'default\'/shipping/options',
+                    value: shippingOptions,
+                },
+            ]);
+        }
+
+        return actions.resolve();
+    }
+
+    private _isAddressSame(address1: CurrentShippingAddress | undefined, address2: CurrentShippingAddress | undefined) {
+        return JSON.stringify(address1) === JSON.stringify(address2);
+    }
+
+    private _transformUSCodes(code: string) {
+       return  UNITED_STATES_CODES.find((state: UnitedStatesCodes) => {
+            return state.name === code && state.abbreviation;
+        });
+    }
+
+    private async _transformToAddress(contact: ShippingAddress) {
+        const countries = this._store.getState().countries.getCountries();
+        const address = {
+            city: contact.city,
+            postalCode: contact.postal_code,
+            countryCode: contact.country_code,
+        };
+        const addressCountry = countries?.find((country: Country) => {
+            return country.code === (contact.country_code || '').toUpperCase();
+        });
+        const stateAddress = addressCountry?.subdivisions.find((region: Region) => {
+
+            return region.code === contact.state?.toUpperCase() || this._transformUSCodes(contact.state);
+        });
+
+        if (stateAddress) {
+            address.postalCode = stateAddress.code;
+        } else {
+            throw new InvalidArgumentError('Invalid Address');
+        }
+
+        return address;
+    }
```

Comment:
We can refactor this method:

```suggestion
    private async _transformToAddress(contact: ShippingAddress) {
        const countries = this._store.getState().countries.getCountries();

        const addressCountry = countries?.find((country: Country) => (
            country.code === (contact.country_code || '').toUpperCase();
        ));
        const stateAddress = addressCountry?.subdivisions.find((region: Region) => (
            region.code === contact.state?.toUpperCase() || region.code === this. _getUSStateByCode(contact.state)?.abbreviation;
        ));

        if (!stateAddress) {
            throw new InvalidArgumentError('Invalid Address');   
        }

        return {
            city: contact.city,
            postalCode: stateAddress.code,
            countryCode: contact.country_code,
        };
    }
```

Reply:
fixed